lib/math_helper.ex

```diff
@@ -59,4 +59,41 @@ defmodule MathHelper do
   """
   @spec log(number(), number()) :: number()
   def log(x, b), do: :math.log(x) / :math.log(b)
+
+  @doc """
+  Returns the byte size of an integer
+
+  ## Examples
+
+      iex> MathHelper.integer_byte_size(0)
+      0
+
+      iex> MathHelper.integer_byte_size(1)
+      1
+
+      iex> MathHelper.integer_byte_size(0xfffffffff)
+      5
+
+  """
+  @spec integer_byte_size(number()) :: number()
+  def integer_byte_size(n) when n == 0, do: 0
+  def integer_byte_size(n), do: byte_size(:binary.encode_unsigned(n))
```

Comment:
What about `n |> :math.log2 |> Kernel./(8) |> :math.ceil |> round`? This should be significantly faster than actually encoding very large numbers.

Reply:
Hmmm, I futzed with this for a while but couldn't get it to work.
I tried:

`n |> :math.log2 |> Kernel./(8) |> :math.ceil |> round`

and

`n |> :math.log2 |> Kernel./(8) |> :math.ceil |> round |> Kernel.+(1)`

And various +1s and -1s all of which failed some common tests. Possibly it's a rounding error?

I agree converting to binary just to get the byte size seems silly and if we are going to be using these a lot optimizations are probably worth it. Alternatively maybe we could do this with `Bitwise` operators?

I'm going to merge as is and we can come back to it. Feel free to take a whack if you like.

Reply:
On second thought: there's a good chance that `:binary.encode_unsigned/1` is not a very expensive operation. Integers are obviously stored in bytes in the first place, and that operation may just wrap those bytes in a different data structure. 👍